plugins/inputs/statsd/running_stats.go

```diff
@@ -69,12 +77,32 @@ func (rs *RunningStats) AddValue(v float64) {
 		// Reached limit, choose random index to overwrite in the percentile array
 		rs.perc[rand.Intn(len(rs.perc))] = v
 	}
+
+	rs.MedLen = len(rs.med)
+	// Need to sort for median
+	sort.Float64s(rs.med)
+	if rs.MedLen < rs.MedLimit {
+		rs.med = append(rs.med, v)
+	} else {
+		// Reached limit, choose random index to overwrite in the median array
+		rs.med[rand.Intn(rs.MedLen)] = v
+	}
 }
 
 func (rs *RunningStats) Mean() float64 {
 	return rs.k + rs.ex/float64(rs.n)
 }
 
+func (rs *RunningStats) Median() float64 {
+	if rs.MedLen == 0 {
+		return 0
+	} else if rs.MedLen%2 == 0 {
+		return (rs.med[rs.MedLen/2-1] + rs.med[rs.MedLen/2]) / 2
+	} else {
+		return rs.med[rs.MedLen/2]
+	}
```

Comment:
I think storing `MedLen` is completely superfluous

```suggestion
	count := len(rs.med)
	
	if count == 0 {
		return 0
	}
	if count%2 == 0 {
		return (rs.med[count/2-1] + rs.med[count/2]) / 2
	}
	return rs.med[count/2]
```

Reply:
changes made